src/canvas/Renderer.tsx

```diff
@@ -315,12 +339,32 @@ export class Renderer implements IRenderer {
     };
 
     private _tick = (): void => {
-        // this._update();
+        if (this._canvas.width === 300 && this._canvas.height === 150) {
```

Comment:
why these values here?

Reply:
This is the default size the canvas is created at, and I want to make sure the update the canvas size when the canvas originally loads, so I check it here. It's kind of a hack, but it works ok.

Reply:
Ok. Why not do that in the init method, so that the check doesn't happen every single tick?

Reply:
Also, I'd add comments on this, since like you said it's kind of a hack.

Reply:
It has to be it _tick because when init is run, the canvas element doesn't actually exist yet. Since, the canvas element comes back null, the resize fails. Init initialized the render class but not necessarily the canvas.

Reply:
I can add a comment about that if you think that would be helpful in the future.

Reply:
I'm still missing something then. The canvas element is created in the Renderer constructor (line 26). We call init in when the CanvasContainer is created in Main.tsx. How can we call init on the Renderer if its constructor hasn't been called?

Reply:
The canvas element isn't appended to the dom when init is called. As we discussed, used a custom event instead so we don't need to check every tick.